### PR TITLE
Juvenile last name

### PR DESCRIPTION
**Pull request by EdwinGuzman**
## Description

Juvenile cards are created without last names! This fixes that.

## Motivation and Context

Fixes [DQ-369](https://jira.nypl.org/browse/DQ-369). The SimplyE mobile clients mark the child's last name in the juvenile account form as optional... so when the user doesn't enter anything, the account has no last name. The assumption is that the child's last name should be the last name of the parent account.

## How Has This Been Tested?

Through Postman:
Created an account with the parent name "Fortitude Patience" (the ILS stores names as "LASTNAME, FIRSTNAME"). When creating a child account with name "Tommy", the full name is now "Tommy PATIENCE". All names get set to uppercase in the ILS so there is a discrepancy to the user since they see the data as it is submitted but not as it is stored.

Screen Shot 2020-09-15 at 6 35 15 PM

## Checklist:

- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/controllers/v0.3/endpoints.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const {
   errorResponseDataWithTag,
 } = require("../../helpers/responses");
 const DependentAccountAPI = require("./DependentAccountAPI");
-const { strToBool } = require("../../helpers/utils");
```
Comment (adriana-alter) [inline api/controllers/v0.3/endpoints.js:20]: What was this doing; why don't we need it anymore?

File: api/helpers/utils.js [hunk 2]
```diff
@@ -28,6 +27,38 @@ const strToBool = (str) => {
   return valsHash[found || str.toLowerCase()];
 };
 
+/**
+ * updateJuvenileName
+ * Update the juvenile's name in case no last named was passed with the
```
Comment (adriana-alter) [inline api/helpers/utils.js:32]: Typo; "last name", not "named"

File: api/helpers/utils.js [hunk 3]
```diff
@@ -28,6 +27,38 @@ const strToBool = (str) => {
...
+const updateJuvenileName = (name, parentArrayName = []) => {
+  const parentsName = parentArrayName[0];
+  // If there's no parent name, then just return the name for the child.
+  if (!parentsName) {
+    return name;
+  }
+
+  let updatedName = name;
+  // If there's no last name, then use the parent's last name. This is a very
+  // basic check that is done by checking if there is a space in the complete
```
Comment (adriana-alter) [inline api/helpers/utils.js:46]: But it's possible for someone's first name to be more than one word...

File: api/controllers/v0.3/endpoints.js (hunk 1 above)
Comment (EdwinGuzman) [reply api/controllers/v0.3/endpoints.js:20]: If the client passes booleans as strings, `"false"` or `"true"`, this converts it to the boolean value, but it wasn't used in this file. Just accidentally imported.

File: api/helpers/utils.js (hunk 3 above)
Comment (EdwinGuzman) [reply api/helpers/utils.js:46]: Yes, but right now we don't take first or last names as separate request inputs. It's just `"name"`. For now, this is fine but I'll make a ticket to make the request accept first and last name which is a larger update than this which is needed soon.

**Review by adriana-alter — APPROVED** [review]

**Pull request merged**